src/VirtoCommerce.CustomerExportImportModule.Data/Validation/ImportContactsValidator.cs

```diff
@@ -1,20 +1,29 @@
 using FluentValidation;
+using Microsoft.AspNetCore.Identity;
 using VirtoCommerce.CustomerExportImportModule.Core.Models;
+using VirtoCommerce.Platform.Core.Common;
+using VirtoCommerce.Platform.Core.Security;
 
 namespace VirtoCommerce.CustomerExportImportModule.Data.Validation
 {
     public sealed class ImportContactsValidator: AbstractValidator<ImportRecord<CsvContact>[]>
     {
-        public ImportContactsValidator()
+        private readonly ICountriesService _countriesService;
+        private readonly SignInManager<ApplicationUser> _signInManager;
+
+        public ImportContactsValidator(ICountriesService countriesService, SignInManager<ApplicationUser> signInManager)
         {
+            _countriesService = countriesService;
+            _signInManager = signInManager;
+
             AttachValidators();
         }
 
         private void AttachValidators()
         {
             RuleFor(importRecords => importRecords).SetValidator(_ => new ImportEntitiesAreNotDuplicatesValidator<CsvContact>());
-            RuleForEach(importRecords => importRecords).SetValidator(new ImportMemberValidator<CsvContact>());
-            RuleForEach(importRecords => importRecords).SetValidator(new ImportContactValidator());
+            RuleForEach(importRecords => importRecords).SetValidator(new ImportMemberValidator<CsvContact>(_countriesService));
+            RuleForEach(importRecords => importRecords).SetValidator(new ImportContactValidator(_signInManager?.UserManager));
```

Comment:
if you want to check _signInManager for null, it better to do in the constructor like:
`_signInManager = signInManager ?? throw new ...;`
instead of passing possible null to the validator.

Reply:
This is workaround for impossiblity to create a mock for it in tests. It can't be null in real system

Reply:
Maybe it'll help https://stackoverflow.com/questions/48189741/mocking-a-signinmanager

Reply:
Refactored